mmdet/models/utils/transformer.py

```diff
@@ -25,6 +29,345 @@
     from mmcv.cnn.bricks.transformer import MultiScaleDeformableAttention
 
 
+class AdaptivePadding(nn.Module):
+    """Applies padding to input (if needed) so that input can get fully covered
+    by filter you specified. It support two modes "same" and "corner". The
+    "same" mode is same with "SAME" padding mode in TensorFlow, pad zero around
+    input. The "corner"  mode would pad zero to bottom right.
+
+    Args:
+        kernel_size (int | tuple): Size of the kernel:
+        stride (int | tuple): Stride of the filter. Default: 1:
+        dilation (int | tuple): Spacing between kernel elements.
+            Default: 1
+        padding (str): Support "same" and "corner", "corner" mode
+            would pad zero to bottom right, and "same" mode would
+            pad zero around input. Default: "corner".
+    Example:
+        >>> kernel_size = 16
+        >>> stride = 16
+        >>> dilation = 1
+        >>> input = torch.rand(1, 1, 15, 17)
+        >>> adap_pad = AdaptivePadding(
+        >>>     kernel_size=kernel_size,
+        >>>     stride=stride,
+        >>>     dilation=dilation,
+        >>>     padding="same")
+        >>> out = adap_pad(input)
+        >>> # padding to divisible by 16
```

Comment:
Why is 16? Whether 16 is kernel size or stride?
According to the formulation `pad_h = max((output_h - 1) * stride_h + (kernel_h - 1) * self.adap_padding.dilation[0] + 1`, if kernel_size is not equal to stride, the padded shape will not be divisible by kernel size or strice.

Reply:
I will remove this comment